tests/contrib/stats/test_localstoragestats.py

```diff
@@ -0,0 +1,98 @@
+from collections import deque
+
+import pytest
+
+from spidermon.contrib.stats.statscollectors import LocalStorageStatsHistoryCollector
+
+
+@pytest.fixture()
+def get_stats_collector(tmpdir):
+    def _stats_collector(crawler):
+        stats_collector = LocalStorageStatsHistoryCollector(crawler)
+        stats_collector._statsdir = tmpdir.strpath
+        return stats_collector
+
+    return _stats_collector
+
+
+def test_spider_has_stats_history_attribute_when_opened_with_collector(
+    get_crawler, get_stats_collector
+):
+    crawler = get_crawler()
```

Comment:
I’m used to the pytest fixture functions defined without the `get_` prefix (e.g. `crawler` instead of `get_crawler`), and when included in the parameters of test functions they are filled with the response value already, and do not need to be called. I am not sure if I am missing something here that explains the difference.

In detail, I see here we are using:

```python
# conftest.py
@fixture
def get_crawler():
    # …

# test_localstoragestats.py
def test_something(get_crawler):
    crawler = get_crawler()
    # …
```

Where I would expect:

```python
# conftest.py
@fixture
def crawler():
    # …

# test_localstoragestats.py
def test_something(crawler):
    # …
```

Reply:
I wanted the fixture to be a factory function for a Crawler instance because for some tests I want to send some extra arguments for the Crawler initialization.

Reply:
A factory function makes a lot of sense. But then I am not sure what’s the benefit of using a fixture. Wouldn’t it work the same if instead of a fixture returning a function you simply define that function and call it from tests?